controllers/datadogslo/controller.go

```diff
@@ -0,0 +1,308 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package datadogslo
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"github.com/DataDog/datadog-operator/controllers/finalizer"
+	"github.com/DataDog/datadog-operator/controllers/utils"
+	ctrUtils "github.com/DataDog/datadog-operator/pkg/controller/utils"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/comparison"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/condition"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/datadog"
+	"github.com/DataDog/datadog-operator/pkg/datadogclient"
+)
+
+const (
+	defaultRequeuePeriod    = 60 * time.Second
+	defaultErrRequeuePeriod = 5 * time.Second
+	defaultForceSyncPeriod  = 60 * time.Minute
+	datadogSLOKind          = "DatadogSLO"
+	datadogSLOFinalizer     = "finalizer.slo.datadoghq.com"
+)
+
+type Reconciler struct {
+	client        client.Client
+	datadogClient *datadogV1.ServiceLevelObjectivesApi
+	datadogAuth   context.Context
+	versionInfo   *version.Info
+	log           logr.Logger
+	recorder      record.EventRecorder
+}
+
+func NewReconciler(client client.Client, ddClient datadogclient.DatadogSLOClient, versionInfo *version.Info, log logr.Logger, recorder record.EventRecorder) *Reconciler {
+	return &Reconciler{
+		client:        client,
+		datadogClient: ddClient.Client,
+		datadogAuth:   ddClient.Auth,
+		versionInfo:   versionInfo,
+		log:           log,
+		recorder:      recorder,
+	}
+}
+
+var _ reconcile.Reconciler = (*Reconciler)(nil)
+
+func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconcile.Result, error) {
+	res, err := r.internalReconcile(req)
+	return res, err
+}
+
+func (r *Reconciler) internalReconcile(req reconcile.Request) (reconcile.Result, error) {
+	logger := r.log.WithValues("datadogslo", req.NamespacedName)
```

Comment:
Seems like the controller has a logger already with a slightly different name https://github.com/DataDog/datadog-operator/pull/807/files#diff-32a274b974ae70f092b3f061d72858382354b8f710d468d80883475162e3920aR31

Reply:
yea I noticed that; I'm not entirely sure why we have this pattern of two loggers but it's present in the DatadogAgent controller as well. when testing I found this logger was a bit more informative to use since it also records the resource name and namespace